csrc/device_lower/analysis/tma.cpp

```diff
@@ -423,12 +421,11 @@ TMAInfo getTMAInfo(LoadStoreOp* ldst) {
         auto g = tma_domain[i].as<ValGroupAndItsGraph>().group;
         tma_g_to_box_g.emplace(g, g);
       }
-    } else {
-      i++;
+      i--;
     }
   }
   // merge contiguous C with SB/CB
-  for (auto i : c10::irange((int64_t)tma_domain.size() - 1)) {
+  for (int64_t i = 0; i < (int64_t)tma_domain.size() - 1; i++) {
```

Comment:
`tma_domain.size()` is changing over time, so we can not use `irange`.

Reply:
Do you mean things like line 436? Surprised that would impact `irange`.

Reply:
Yes. `irange` compute the size at the beginning, and use it for the entire loop. It does not dynamically check if size has changed.

Reply:
Ah, I missed you also added `i--`. Now I got it.